USE_FULL_ASSERT results in failed build due to missing stm32_assert.h

when including -DUSE_FULL_ASSERT in build-opt.h, the build fails with error

```
In file included from C:\Users\xXxXxXx\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.7.1\libraries\SrcWrapper\src\LL\stm32yyxx_ll_adc.c:32:
C:\Users\xXxXxXx\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.7.1\system/Drivers/STM32L4xx_HAL_Driver/Src/stm32l4xx_ll_adc.c:25:10: fatal error: stm32_assert.h: No such file or directory
   25 | #include "stm32_assert.h"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
exit status 1

Compilation error: exit status 1
```

--- a/cores/arduino/stm32/stm32_assert.h
+++ b/cores/arduino/stm32/stm32_assert.h
@@ -1,0 +1,10 @@
+/* Define to prevent recursive inclusion -------------------------------------*/
+#ifndef STM32_ASSERT_H
+#define STM32_ASSERT_H
+
+/*
+ * Header file required by LL layer but as USE_HAL_DRIVER is defined
+ * assert definitions are provided by stm32yyxx_hal_conf_default
+ */
+
+#endif /* STM32_ASSERT_H */

--- a/cores/arduino/stm32/stm32_def.h
+++ b/cores/arduino/stm32/stm32_def.h
@@ -133,6 +133,8 @@ void SystemClock_Config(void);
 #if !defined(_Error_Handler)
 #define _Error_Handler(str, value) \
   while (1) {\
+    (void)str;\
+    (void)value;\
   }
 #endif
 #if !defined(Error_Handler)

--- a/libraries/SrcWrapper/src/stm32/stm32_def.c
+++ b/libraries/SrcWrapper/src/stm32/stm32_def.c
@@ -20,6 +20,20 @@ WEAK void _Error_Handler(const char *msg, int val)
 }
 #endif
 
+#ifdef  USE_FULL_ASSERT
+/**
+  * @brief  Reports the name of the source file and the source line number
+  *         where the assert_param error has occurred.
+  * @param  file: pointer to the source file name
+  * @param  line: assert_param error line source number
+  * @retval None
+  */
+WEAK void assert_failed(uint8_t *file, uint32_t line)
+{
+  _Error_Handler((const char *)file, line);
+}
+#endif /* USE_FULL_ASSERT */
+
 #ifdef __cplusplus
 }
 #endif

